loadtest/app_test.go

```diff
@@ -0,0 +1,62 @@
+package loadtest
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func getNewTestSampleSpace() []operation {
+	var sampleSpace []operation
+	sampleSpace = append(sampleSpace, operation{
+		probability: 0.2,
+	})
+	sampleSpace = append(sampleSpace, operation{
+		probability: 0.3,
+	})
+	return sampleSpace
+}
+
+var _ = Describe("Load Test Application", func() {
+	Describe("normalizeSampleSpace()", func() {
+		It("Should change slice contents", func() {
+			const pSum float64 = 0.5
+			var sampleSpace []operation
+			sampleSpace = append(sampleSpace, operation{
+				probability: pSum,
+			})
+			normalizeSampleSpace(sampleSpace, pSum)
+			Expect(sampleSpace[0].probability).To(Equal(1.0))
+		})
+	})
+
+	Describe("getRandomOperationFromSampleSpace()", func() {
+		It("Should return first operation", func() {
+			sampleSpace := getNewTestSampleSpace()
+			operation, err := getRandomOperationFromSampleSpace(sampleSpace, 0.15)
+			Expect(operation.probability).To(Equal(sampleSpace[0].probability))
+			Expect(err).To(BeNil())
+		})
+
+		It("Should return second operation", func() {
+			sampleSpace := getNewTestSampleSpace()
+			operation, err := getRandomOperationFromSampleSpace(sampleSpace, 0.25)
+			Expect(operation.probability).To(Equal(sampleSpace[1].probability))
+			Expect(err).To(BeNil())
+		})
+
+		It("Should return last operation", func() {
+			sampleSpace := getNewTestSampleSpace()
+			operation, err := getRandomOperationFromSampleSpace(sampleSpace, 0.5)
+			Expect(operation.probability).To(Equal(sampleSpace[1].probability))
+			Expect(err).To(BeNil())
+		})
+
+		It("Should return an error because dice is larger than one", func() {
```

Comment:
When would this happen?

Reply:
Float errors =(

Reply:
When normalizing the space? How can we ensure that this won't happen? I'm not comfortable with raising an error in this scenario.

Reply:
You're right. I'll just return the first operation in this case. Despite being possible to happen, those errors should happen almost never. Thanks!